### DISCUSSION TITLE
Development Environment for OSX

### DISCUSSION THREAD

**Discussion by robotlovecoffee** [ordinals/ord#3783] [Q&A]
I have a ORD running locally using vscode and I'm able to debug and step thru code but cannot close a session without causing the DB to get corrupted.   Stopping does send a SIGNET but that does not seem to be picked up to allow the code to close gracefully.  

What environments are people using to debug and test ORD locally on OSX?

**Comment by raphjaph** [comment]
Our handler only listens to `SIGINT` and `SIGTERM`. I use `nvim`, `dbg!()` and `cargo run` for debugging so no idea how to set it properly for VSCode

**Comment by robotlovecoffee** [comment]
Thanks for the answer and clarification.  I will see how I can send those and see if I can get it working.  Failing that I will look at the environment that you are using.  I know that I'm an edge case but I do think trying to document how to setup a dev environment would be helpful for those of us that are coming from a different stack completely.  If I get stable I will document it.

**Comment by robotlovecoffee** [comment]
Sorry I also made a typo when I posted, stopping should be sending a SIGINT from vscode but I think I need to check the extension that is doing the debugging as perhaps it does not do this correctly.

I was trying to do this but it was not working:

```
ps -eaf |grep <Proc-Name>   //  find the PID

kill -s SIGINT PID_OF_PROCESS
```

I will keep working on this to see if I can get a stable dev environment.
